lib/chef/provider/powershell_script.rb

```diff
@@ -24,71 +24,153 @@ class PowershellScript < Chef::Provider::WindowsScript
 
       provides :powershell_script, os: "windows"
 
+      def initialize (new_resource, run_context)
+        super(new_resource, run_context, '.ps1')
+        add_exit_status_wrapper
+      end
+
+      def action_run
+        valid_syntax = validate_script_syntax!
+        super if valid_syntax
```

Comment:
See my reply below to @ksubrama. Basically you can get true or false or a exception raised. False means it wasn't valid, but the user specified a return value that should suppress the error. This is how existing script resources work. In that case, I'd like to avoid running the script again (hence the conditional `super`). It's probably not necessary, but it's easier to to reason about I guess in saying that if the syntax is not valid, we won't bother trying to run it.